pinot-core/src/main/java/org/apache/pinot/core/transport/QueryRouter.java

```diff
@@ -195,16 +197,18 @@ void markQueryDone(long requestId) {
     _asyncQueryResponseMap.remove(requestId);
   }
 
-  private InstanceRequest getInstanceRequest(long requestId, BrokerRequest brokerRequest, List<String> segments) {
+  private InstanceRequest getInstanceRequest(long requestId, BrokerRequest brokerRequest,
+      Pair<List<String>, List<String>> segments) {
     InstanceRequest instanceRequest = new InstanceRequest();
     instanceRequest.setRequestId(requestId);
     instanceRequest.setQuery(brokerRequest);
     Map<String, String> queryOptions = brokerRequest.getPinotQuery().getQueryOptions();
     if (queryOptions != null) {
       instanceRequest.setEnableTrace(Boolean.parseBoolean(queryOptions.get(CommonConstants.Broker.Request.TRACE)));
     }
-    instanceRequest.setSearchSegments(segments);
+    instanceRequest.setSearchSegments(segments.getLeft());
     instanceRequest.setBrokerId(_brokerId);
+    instanceRequest.setOptionalSegments(segments.getRight());
```

Comment:
What is the contract here when there is no optional segments? Should it be an empty list or `null`? Let's fix the contract and make sure it is properly applied

Reply:
I'll make contract as just using `null` if no optional segments